index.js

```diff
@@ -56,37 +58,41 @@ const query = `
 	}
 `;
 
-let responseText = '[]';
 let responseETag = '';
 
-async function fetchRepos() {
-	const {body} = await graphqlGot('api.github.com/graphql', {
-		query,
-		token
-	});
-
-	const repos = body.user.repositories.nodes.map(repo => ({
-		...repo,
-		stargazers: repo.stargazers.totalCount,
-		forks: repo.forks.totalCount
-	}));
-	responseText = JSON.stringify(repos);
-	responseETag = etag(responseText);
-}
-
-setInterval(fetchRepos, ONE_DAY);
-fetchRepos();
-
-module.exports = (request, response) => {
+module.exports = async (request, response) => {
 	controlAccess()(request, response);
-	response.setHeader('cache-control', cache);
-	response.setHeader('etag', responseETag);
 
 	if (request.headers.etag === responseETag) {
 		response.statusCode = 304;
 		response.end();
 		return;
 	}
 
-	response.end(responseText);
+	try {
+		const {body} = await graphqlGot('api.github.com/graphql', {
+			query,
+			token: GITHUB_TOKEN
+		});
+
+		const repos = JSON.stringify(
+			body.user.repositories.nodes.map(repo => ({
+				...repo,
+				stargazers: repo.stargazers.totalCount,
+				forks: repo.forks.totalCount
+			}))
+		);
+
+		responseETag = etag(repos);
```

Comment:
Now that I think of it, will we ever reach this again? Wouldn't the `ETag` always match `If-None-Match` after the first request.

Reply:
I removed the `ETag` matching altogether since I think `s-maxage` should be enough.